github-actions/src/main/scala/org/typelevel/sbt/gha/WorkflowJob.scala

```diff
@@ -86,6 +88,7 @@ object WorkflowJob {
       container: Option[JobContainer] = None,
       environment: Option[JobEnvironment] = None,
       concurrency: Option[Concurrency] = None,
+      outputs: Map[String, String] = Map(),
```

Comment:
This is the binary-breaking change. To restore binary-compatibility, we need to keep an overload of `apply` with the old signature, that can delegate to the new one.

It's also a source-breaking change: `outputs` should be added as the final parameter in the list.